src/control/cmd/daos_agent/mgmt_rpc.go

```diff
@@ -27,14 +28,15 @@ import (
 // Management Service proxy, handling dRPCs sent by libdaos by forwarding them
 // to MS.
 type mgmtModule struct {
-	log        logging.Logger
-	sys        string
-	ctlInvoker control.Invoker
-	aiCache    *attachInfoCache
-	numaAware  bool
-	netCtx     context.Context
-	mutex      sync.Mutex
-	monitor    *procMon
+	log         logging.Logger
+	sys         string
+	ctlInvoker  control.Invoker
+	cacheMutex  sync.Mutex
```

Comment:
Ah, yeah, I went back and forth on this a bit. I intended to remove the internal mutexes in favor of wrapping sets of operations, checking IsCached() and then Cache() if needed. I suppose the race in that case doesn't matter that much, as it would only happen once during agent uptime.